lib/shopify_transporter/shopify/customer.rb

```diff
@@ -57,12 +57,30 @@ def to_csv
       ).freeze
 
       def top_level_row_values
-        base_hash.merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES)).values
+        default_address.merge(top_level_attributes).values_at(*Customer.columns)
+      end
+
+      def top_level_attributes
+        base_hash
+          .merge(record_hash.slice(*TOP_LEVEL_ATTRIBUTES))
+          .delete_if { |_, v| v.nil? }
+      end
+
+      def default_address
+        if at_least_one_address?
+          record_hash['addresses'][0].slice(*ADDRESS_ATTRIBUTES)
+        else
+          {}
+        end
+      end
+
+      def at_least_one_address?
+        record_hash.dig('addresses').present? && record_hash.dig('addresses')[0].present?
       end
 
       def address_row_values
-        return [] unless record_hash['addresses']
-        record_hash['addresses'].map do |address_hash|
+        return [] unless record_hash['addresses'].present?
+        record_hash['addresses'].drop(1).map do |address_hash|
```

Comment:
The `drop(1)` feels like a hack. We should probably make sure it's the right amount of addresses before we map over it.

Reply:
I've added check in the beginning to make sure the array is of size 2 at least. Does this make `drop(1)`seem less like a hack?

Reply:
Yes, I'm more okay with it now. 👍